packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -33,33 +34,59 @@ export function FileUploader({
     addTargetFiles,
     showPreviewer,
     setFiles,
-  } = useFileUploader();
+    fileStatuses,
+    setFileStatuses,
+    setFileSizeErrors,
+  } = useFileUploader(maxSize, acceptedFileTypes, multiple);
+
   const [allFileNames, setAllFileNames] = useState<string[]>([]);
-  const [fileStatuses, setFileStatuses] = useState<FileStatuses>([]);
   const fileStatusesRef = useRef<FileStatuses>([]);
   // File Previewer global states
   const [isLoading, setLoading] = useState(false);
   const [isSuccess, setSuccess] = useState(false);
   const [percentage, setPercentage] = useState(0);
+  // Tracker states
+  const [isEditingNames, setIsEditingNames] = React.useState<boolean[]>([]);
 
   useEffect(() => {
     if (fileStatuses.length === 0) return;
-    const success = fileStatuses.every((status) => status?.percentage === 100);
-    setSuccess(success);
-    setLoading(!success);
+    // Sets variable when all downlaods are complete
+    const complete = fileStatuses.every((status) => status?.percentage === 100);
+    setSuccess(complete);
 
+    // Creates aggregate percentage to show during downloads
     const percentage =
       fileStatuses.reduce((prev, curr) => prev + (curr?.percentage ?? 0), 0) /
       fileStatuses.length;
     setPercentage(Math.floor(percentage));
+
+    // Loading ends when all files are at 100%
+    if (Math.floor(percentage) === 100) {
+      setLoading(false);
+    }
   }, [fileStatuses]);
 
   useEffect(() => {
     setShowPreviewer(isPreviewerVisible);
   }, [setShowPreviewer, isPreviewerVisible]);
 
   useEffect(() => {
-    setAllFileNames(files.map((file) => file.name));
+    setAllFileNames((allFileNames) => {
+      if (allFileNames.length !== 0) {
+        // only add the new files to allFileNames
+        const diff = files.length - allFileNames.length;
+        const arrayOfNames: string[] = [];
+
+        for (let i = 0; i < diff; i++) {
+          arrayOfNames[i] = files[i].name;
+        }
+
+        return [...arrayOfNames].concat(allFileNames);
+      }
+      return files.map((file) => {
+        return file.name;
+      });
```

Comment:
The idea is that we need to maintain the `allFileNames` array so the indexes match between it, and the other arrays. If the `allFileNames` array is empty than we fill it with all the `file.names`. We do this since `file.name` cannot be edited (security rule when working with JavaScrip files grabbed from an input), and was easier to maintain the file names in a separate array.